geode-core/src/main/java/org/apache/geode/cache/query/internal/QueryConfigurationServiceImpl.java

```diff
@@ -0,0 +1,146 @@
+      "Please use the UnrestrictedMethodAuthorizer implementation of MethodInvocationAuthorizer " +
+      "instead";
+
+  public QueryConfigurationServiceImpl() {
+    ALLOW_UNTRUSTED_METHOD_INVOCATION = Boolean.parseBoolean(
+        System.getProperty(GEMFIRE_PREFIX + "QueryService.allowUntrustedMethodInvocation"));
+  }
+
+  public static MethodInvocationAuthorizer getNoOpAuthorizer() {
+    // A no-op authorizer, allow method invocation
+    return ((Method m, Object t) -> true);
+  }
+
+  @Override
+  public boolean init(Cache cache) {
+    if (cache == null) {
+      throw new IllegalArgumentException("cache must not be null");
+    }
+
+    if (System
+        .getProperty(GEMFIRE_PREFIX + "QueryService.allowUntrustedMethodInvocation") != null) {
+      logger.warn(DEPRECATION_WARNING);
+    }
+
+    if (isSecurityDisabled((InternalCache) cache) || ALLOW_UNTRUSTED_METHOD_INVOCATION) {
+      this.authorizer = getNoOpAuthorizer();
+    } else {
+      this.authorizer = new RestrictedMethodAuthorizer(cache);
+    }
+    return true;
+  }
+
+  @Override
+  public Class<? extends CacheService> getInterface() {
+    return QueryConfigurationService.class;
+  }
+
+  @Override
+  public CacheServiceMBeanBase getMBean() {
+    return null;
+  }
+
+  @Override
+  public void close() {
+
+  }
+
+  @Override
+  public MethodInvocationAuthorizer getMethodAuthorizer() {
+    return authorizer;
+  }
+
+  @Override
+  public void updateMethodAuthorizer(Cache cache, QueryMethodAuthorizerCreation creation)
+      throws QueryConfigurationServiceException {
+    updateMethodAuthorizer(cache, creation.getClassName(), creation.getParameters());
+  }
+
+  @Override
+  public void updateMethodAuthorizer(Cache cache, String className, Set<String> parameters)
+      throws QueryConfigurationServiceException {
+    if (isSecurityDisabled((InternalCache) cache) || ALLOW_UNTRUSTED_METHOD_INVOCATION) {
+      return;
+    }
+
+    try {
+      if (className.equals(RestrictedMethodAuthorizer.class.getName())) {
+        this.authorizer = new RestrictedMethodAuthorizer(cache);
+      } else if (className.equals(UnrestrictedMethodAuthorizer.class.getName())) {
+        this.authorizer = new UnrestrictedMethodAuthorizer(cache);
+      } else if (className.equals(JavaBeanAccessorMethodAuthorizer.class.getName())) {
+        this.authorizer = new JavaBeanAccessorMethodAuthorizer(cache, parameters);
+      } else if (className.equals(RegExMethodAuthorizer.class.getName())) {
+        this.authorizer = new RegExMethodAuthorizer(cache, parameters);
+      } else {
+        @SuppressWarnings("unchecked")
+        Class<MethodInvocationAuthorizer> userClass =
+            (Class<MethodInvocationAuthorizer>) ClassPathLoader.getLatest().forName(className);
+        Constructor<MethodInvocationAuthorizer> userConstructor =
+            userClass.getDeclaredConstructor(Cache.class, Set.class);
+        this.authorizer = userConstructor.newInstance(cache, parameters);
```

Comment:
The above implies that, whenever our users want to provide their own `MethodInvocationAuthorizer`, the class must provide at least one constructor receiving a `Cache` and a `Set` as parameters, right?. What's the reasoning behind this change?, shouldn't we require a constructor with no parameters and invoke `Declarable.init()` afterwards as we do with other customisations provided by customers?.
Am I missing something?.

Reply:
good question, I think you mean initialize(Cache cache, Properties properties)?

Reply:
@jhuynh1 : you're right, the old `init` method is deprecated in favour of `initialize` 👍 .

Reply:
After some investigation we discovered that implementing declarable would require significant changes to the rest of this feature and would also introduce other challenges around configuration that would need to be solved. We decided to change this constructor to an init method that mirrors the pattern of the Declarable interface without actually using the interface itself - this should make it easier for users to implement custom authorizers similar to how they implement other custom classes within Geode.